paimon-core/src/main/java/org/apache/paimon/utils/SnapshotManager.java

```diff
@@ -352,13 +377,16 @@ public Snapshot traversalSnapshotsFromLatestSafely(Filter<Snapshot> checker) {
         return null;
     }
 
-    private @Nullable Long findLatest() throws IOException {
-        Path snapshotDir = snapshotDirectory();
+    private @Nullable Long findLatest(String branchName) throws IOException {
+        Path snapshotDir =
+                branchName.equals(DEFAULT_MAIN_BRANCH)
+                        ? snapshotDirectory()
```

Comment:
This piece of code appeared many times, could you provide a method to express this directory logic simply?

Reply:
Good idea！ I will fix it.